src/extract.ts

```diff
@@ -0,0 +1,17 @@
+import { left, right } from 'fp-ts/lib/TaskEither'
+import { ApplicationError, StatusCodes } from './http'
+
+export function extractSource(body: any) {
```

Comment:
It should not be `any`. For sure there is a type for that in the opensearch client lib.